src/common.ts

```diff
@@ -20,6 +20,17 @@ export function openAuthenticationWindow(
   return url
 }
 
+export function validateBaseURL(url?: string): string {
+  if (typeof url !== 'string') {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  const match = url.match(/^(https:\/\/[\w-]+(\.[\w-]+)*(:\d{1,5})?)\/?$/)
+  if (!match) {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  return match[1]
+}
+
```

Comment:
I think the `validateUrl` function in `BaseConnector` can be used here

Reply:
See below